### ISSUE TITLE
Atom typing regression in EEFSOLV

### ISSUE THREAD

**Issue by tlhr** [plumed/plumed2#589]
In systems with charged C termini and OT1 / OT2 atom types, EEFSOLV throws the following error:
```
+++ PLUMED error
+++ at PDB.cpp:254, function std::string PLMD::PDB::getAtomName(PLMD::AtomNumber) const
+++ message follows +++
Name of atom 112 not found
```
I believe this is due to #522, as the code relies on receiving an empty string instead of an error message when referring to nonexistent atoms:

```cpp
// Check for terminal COOH or COO- (different atomtypes & parameters!)
if (moldat[0]->getAtomName(atoms[i]) == "OT1" || moldat[0]->getAtomName(atoms[i]) == "OXT") {
  // We create a temporary AtomNumber object to access future atoms
  unsigned ai = atoms[i].index();
  AtomNumber tmp_an;
  tmp_an.setIndex(ai + 2);
  if (moldat[0]->getAtomName(tmp_an) == "HT2") {
    // COOH
    Atype = "OB";
  } else {
    // COO-
    Atype = "OC";
  }
  cter = true;
}
```

I'm happy to write a fix for this, but I'm not quite sure what the best way of doing it is. Maybe adding something along the lines of `bool SetupMolInfo::AtomExists(AtomNumber a)`? Or throwing an iteration exception, which could be caught in the code above?

**Comment by carlocamilloni** [comment]
I don't understand, the error suggests that atom 112 is not included in the PDB file given in input to MOLINFO, or is there something else?

**Comment by tlhr** [comment]
@carlocamilloni This is what the end of the PDB file looks like:
```
ATOM    108  HG  SER     9     -26.270  -4.330  14.510  1.00  0.00
ATOM    109  C   SER     9     -24.110  -4.600  17.630  1.00  0.00
ATOM    110  OT1 SER     9     -25.040  -4.470  17.970  1.00  0.00
ATOM    111  OT2 SER     9     -23.400  -5.460  18.160  1.00  0.00
END
```
So in the iteration for atom 110 it's searching for atom 112 (HT1, `ai + 2`) but not finding it. Before the fix for #522 `getAtomName()` just returned an empty string, now it throws an exception. This means we never reach the `else` clause to assign the correct atom type.

**Comment by carlocamilloni** [comment]
ok I see, then yes I think have a "bool" action in the PDB tools to check for an atom number may be good solution. There already is an action bool PDB::checkForAtom( const std::string& name ) but checking for the number would be better in this case
